src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -164,6 +167,17 @@ public string StorageConnectionString
         /// </summary>
         public TimeSpan? FunctionTimeout { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether function invocations will timeout when
+        /// a <see cref="FunctionTimeout"/> is specified and a debugger is attached. False by default.
+        /// </summary>
+        public bool TimeoutWhileDebugging { get; set; }
+
+        /// <summary>
+        /// When true, an exception is thrown when a function timeout expires.
+        /// </summary>
+        public bool ThrowExceptionOnFunctionTimeout { get; set; }
```

Comment:
Yeah, I think we need to do something, otherwise mixing the two models doesn't work well. If I've implemented cancellation tokens, I'd want those to be cancelled giving me a chance to clean up before the process goes down.